consensus/src/marshal/actor.rs

```diff
@@ -534,6 +532,33 @@ impl<
                                 }
                             }
                         }
+                        Message::SetSyncFloor { height } => {
+                            if let Some(m_height) = self.application_metadata.get(&LATEST_KEY) {
+                                if *m_height >= height {
+                                    debug!(height, "sync floor not updated, lower than existing");
+                                    continue;
+                                }
+                            }
+
+                            if let Err(e) = self
+                                .application_metadata
+                                .put_sync(LATEST_KEY.clone(), height)
+                                .await
+                            {
+                                error!(?e, height, "failed to update sync floor");
+                                return;
+                            }
+                            self.last_processed_height = height;
+                            let _ = self.processed_height.try_set(self.last_processed_height);
+
+                            // Cancel any existing requests below the new sync floor.
+                            resolver.retain(Request::<B>::Finalized { height }.predicate()).await;
```

Comment:
Ideally we could reuse similar code to `self.handle_block_processed(height)` to share logic there, but I guess firstly we don't have the block commitment, and secondly we know we might be skipping blocks and therefore have to do extra stuff like setting the `pending_ack` back to `None`